tests/rip7560/test_validation_rules.py

```diff
@@ -110,22 +123,16 @@ def get_build_func(entity, rule):
         # Not implemented yet
         pass
     if rule.find("init_code") > 0:
-        build_func = with_initcode(build_func)
+        # STO-022 factory must be staked
+        if entity == SENDER and assert_func is assert_ok:
```

Comment:
I can't say I understand this logic.
I don't see a code where you use unstaked factory - whcih means there are test scenarios you can't test

Reply:
Almost all tests use unstaked factory

Reply:
what I mean: I look at the list of test cases which describe the preconditions for each test.
from this code I understand that sometimes the factory is staked, and sometime it doesn't (that is, when the entity-under-test is not the factory itself)
but from a test case itself, I can't tell the "preconditions"